gnucash/report/report-system/test/test-report-html.scm

```diff
@@ -0,0 +1,217 @@
+(use-modules (gnucash gnc-module))
+
+(gnc:module-begin-syntax (gnc:module-load "gnucash/app-utils" 0))
+
+(use-modules (gnucash engine test test-extras))
+(use-modules (gnucash report report-system))
+(use-modules (srfi srfi-64))
+(use-modules (gnucash engine test srfi64-extras))
+
+(define (run-test)
+    (test-runner-factory gnc:test-runner)
+    (test-begin "Testing/Temporary/test-report-html") ;; if (test-runner-factory gnc:test-runner) is commented out, this
+                                                      ;; will create Testing/Temporary/test-report-html.log
+    (test-assert "HTML Document Definition" (test-check1))
+    (test-assert "HTML Objects Definition for literals" (test-check2))
+    (test-assert "HTML Text Object" (test-check3))
+    (test-assert "HTML Table Cell" (test-check4))
+    (test-assert "HTML Table" (test-check5))
+    (test-end "Testing/Temporary/test-report-html")
+)
+
+;; -----------------------------------------------------------------------
+
+(define (test-check1)
+  (let (
+         (test-doc (gnc:make-html-document))
+       )
+    (and
+      (gnc:html-document? test-doc)
+      (not (gnc:html-document-style-sheet test-doc))
+      (null? (gnc:html-document-style-stack test-doc))
+      (gnc:html-style-table? (gnc:html-document-style test-doc))
+      (not (gnc:html-document-style-text test-doc))
+      (string-null? (gnc:html-document-title test-doc))
+      (not (gnc:html-document-headline test-doc))
+      (null? (gnc:html-document-objects test-doc))
+      (string=? (gnc:html-document-render test-doc)
+        "<html>\n<head>\n<meta http-equiv=\"content-type\" content=\"text/html; charset=utf-8\" />\n<title>\n</title></head><body></body>\n</html>\n"
```

Comment:
I'd think guile does allow multiline strings which are definitely easier to read, and also if the html is amended in code it'll be much easier to make small amendments to this test string.

Reply:
As far as I can see this is handled in the next commit.